gpytorch/lazy/lazy_tensor.py

```diff
@@ -1028,7 +1028,14 @@ def inv_quad_logdet(self, inv_quad_rhs=None, logdet=False, reduce_inv_quad=True)
             from .chol_lazy_tensor import CholLazyTensor
             from .triangular_lazy_tensor import TriangularLazyTensor
 
-            cholesky = CholLazyTensor(TriangularLazyTensor(self.cholesky()))
+            # if the root decomposition has already been computed and is triangular we can use it instead
+            # of computing the cholesky.
+            if _is_in_cache_ignore_args(self, "root_decomposition"):
+                root = self.root_decomposition().root
+                if isinstance(root, TriangularLazyTensor):
```

Comment:
mhh looksl like we'd get an `UnboundLocalError` if this is not true?

Reply:
which is a reasonably passive aggressive way of saying that we should probably have test coverage here :)